### PR TITLE
Swraster

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: skiko/src/jvmMain/cpp/linux/drawlayer.cc [hunk 1]
```diff
@@ -17,12 +17,16 @@ typedef GLXContext (*glXCreateContextAttribsARBProc)(Display *, GLXFBConfig, GLX
 extern "C" jboolean Skiko_GetAWT(JNIEnv *env, JAWT *awt);
 
 JavaVM *jvm = NULL;
+static const int RASTER = 1;
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/linux/drawlayer.cc:20]: Must not be duplicated.

File: skiko/src/jvmMain/cpp/linux/drawlayer.cc [hunk 2]
```diff
@@ -155,16 +159,36 @@ extern "C"
...
+        LayerHandler *layer = new LayerHandler();
+        layerStorage->insert(layer);
+
+        jclass canvasClass = env->GetObjectClass(canvas);
+        static jmethodID renderApiMethod = NULL;
+        if (!renderApiMethod)
+        {
+            renderApiMethod = env->GetMethodID(canvasClass, "getApi", "()I");
+        }
+        if (NULL == renderApiMethod)
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/linux/drawlayer.cc:172]: Why not pass renderer API as an argument of `updateLayer` and do management on Kotlin side.

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/GraphicsApi.kt [hunk 3]
```diff
@@ -1,5 +1,9 @@
 package org.jetbrains.skiko
 
-enum class GraphicsApi {
-    UNKNOWN, OPENGL, VULKAN, METAL
+object GraphicsApi {
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/GraphicsApi.kt:3]: Why this change, enum can have arbitrary int fields, i.e. 
```
enum class GraphicsApi(val id: Int) {
    UNKNOWN(0), OPENGL(1), VULKAN(2), METAL(3)
}
```

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt [hunk 4]
```diff
@@ -63,8 +63,9 @@ object Library {
     // This function doesn't actually belong to this file.
     private fun miscSystemInit() {
         // we have to set this property to avoid render flickering.
-        System.setProperty("sun.awt.noerasebackground", "true")
+        System.setProperty("sun.awt.noerasebackground", "false")
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt:66]: Why this change?

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt [hunk 5]
```diff
@@ -63,8 +63,9 @@ object Library {
...
         System.setProperty("skija.staticLoad", "false")
+        System.setProperty("skiko.renderApi", "OPENGL")
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt:68]: You set property unconditionally here, why?

File: skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.kt [hunk 6]
```diff
@@ -73,7 +84,19 @@ open class SkiaLayer() : HardwareLayer() {
             }
 
             renderer?.onRender(canvas!!, width, height)
-            context!!.flush()
+
+            if (api == GraphicsApi.RASTER) {
+                var img: BufferedImage?
+                val bais = ByteArrayInputStream(surface!!.makeImageSnapshot().encodeToData()!!.bytes)
```
Comment (olonho) [inline skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.kt:90]: Looks like very inefficient solution, cannot we instead get NIO byte buffer wrapping image content instead?

File: skiko/src/jvmMain/objectiveC/macos/drawlayer.m [hunk 7]
```diff
@@ -285,19 +299,34 @@ JNIEXPORT void JNICALL Java_org_jetbrains_skiko_HardwareLayer_updateLayer(JNIEnv
 
         LayerHandler *layersSet = [[LayerHandler alloc] init];
 
+        // render api
+        jclass canvasClass = (*env)->GetObjectClass(env, canvas);
+        static jmethodID renderApiMethod = NULL;
+        if (!renderApiMethod)
+        {
+            renderApiMethod = (*env)->GetMethodID(env, canvasClass, "getApi", "()I");
```
Comment (olonho) [inline skiko/src/jvmMain/objectiveC/macos/drawlayer.m:307]: Same, rendering API shall be passed here, not retrieved.

**Comment by Rsedaikin** [comment]
created new branch -> raster
